File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], source: int, destination: int) -> bool:
        def traversal(node):
            if node==destination:
                return True
            visited[node]=True
            for neigh in adjlist[node]:
                if visited[neigh]==False:
                    if traversal(neigh):
                        return True
            return False
        #-------------------------------
        adjlist=[[] for r in range(n)]

        for u,v in edges:
            adjlist[u].append(v)
            adjlist[v].append(u)

        visited=[False]*n
        return traversal(source)

File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.java
class Solution {

    public boolean dfs(List<List<Integer>> adj,boolean[] visited, int src,int dest){
        if(src==dest){
            return true;
        }
        visited[src]=true;
        for(int neigh: adj.get(src)){
            if(visited[neigh]==false){
                if(dfs(adj,visited,neigh,dest)==true){
                    return true;
                }
            }
        }
        return false;

    }

    public boolean validPath(int n, int[][] edges, int source, int destination) {
        List<List<Integer>> adj=new ArrayList<>();
        for(int i=1;i<=n;i++){
            adj.add(new ArrayList<>());
        }

        for(int[] edge: edges){
            int u=edge[0];
            int v=edge[1];

            adj.get(u).add(v);
            adj.get(v).add(u);
        }

        boolean[] visited=new boolean[n];

        return dfs(adj,visited,source,destination);
    }
}